client/src/process.js

```diff
@@ -0,0 +1,213 @@
+/* eslint-disable no-undef */
+const baseURL = 'http://localhost:10000';
+
+/**
+ * Function displays the overlay
+ */
+
+export function showOverlay() {
+    document.getElementById('overlay').style.display = 'block';
+}
+
+/**
+ * Function hides the overlay
+ */
+
+export function hideOverlay() {
+    document.getElementById('overlay').style.display = 'none';
+}
+
+/**
+ * Function displays the loading spinner and overlay
+ */
+
+export function showSpinner() {
+    showOverlay();
+    document.getElementById('loadingSpinner').style.display = 'block';
+}
+
+/**
+ * Function hides the loading spinner and overlay
+ */
+
+export function hideSpinner() {
+    hideOverlay();
+    document.getElementById('loadingSpinner').style.display = 'none';
+}
+
+/**
+ * Function is called when the user clicks on the "Update" button
+ * Updates the network graph with the user ID entered in the input box
+ */
+
+export function updateID() {
+    console.log('Update ID called');
+    showSpinner();
+
+    let userID = document.getElementById('idInput').value;
+
+    if (!/^\d+$/.test(userID)) {
```

Comment:
Move user ID verification to its own function?

Reply:
yeah and as I mentioned before, this can be a easy funtion to unit test

Reply:
Yep, sounds good

Reply:
*Sorry, to clarify I meant creating a function which just verifies a userID. i.e. a function that accepts a string and returns true/false on whether it's valid. This would be a lot easier to unit test as you don't need to mock the html you can just test with a few different strings, and it can be reused if we need to verify the userID anywhere else. I think leaving the document.getElementById('idInput').value in the main function is fine bc. the function is specifically for the case of a user pressing the update button (and that way you only have to call it once).

Reply:
Modified